src/Exizent.ComponentModel.DataAnnotations/MaxPrecisionAttribute.cs

```diff
@@ -0,0 +1,32 @@
+namespace Exizent.ComponentModel.DataAnnotations;
+
+public class MaxPrecisionAttribute : ValidationAttribute
+{
+    public int MaxPrecision { get; }
+
+    public MaxPrecisionAttribute(int maxPrecision)
+    {
+        MaxPrecision = maxPrecision;
+    }
+
+    protected override ValidationResult? IsValid(object? value, ValidationContext validationContext)
```

Comment:
You should prob just override the [IsValid(Object)](https://docs.microsoft.com/en-us/dotnet/api/system.componentmodel.dataannotations.validationattribute.isvalid?view=net-6.0#system-componentmodel-dataannotations-validationattribute-isvalid(system-object)) method and then setup the ErrorMessage in the ctor so that people can override the message too.